### PR TITLE
Move mimaSettings back to scalaModuleSettings

### PR DESCRIPTION
**Pull request by lrytz**
Running MiMa makes sense also for scala-js artifacts.

### CODE REVIEW

File: src/main/scala/ScalaModulePlugin.scala [hunk 1]
```diff
@@ -143,11 +143,11 @@ object ScalaModulePlugin extends AutoPlugin {
         </developer>
       </developers>
     )
-  )
+  ) ++ mimaSettings
 
   lazy val scalaModuleSettingsJVM: Seq[Setting[_]] = Seq(
     mimaPreviousVersion := None
-  ) ++ mimaSettings ++ scalaModuleOsgiSettings
+  ) ++ scalaModuleOsgiSettings
```
Comment (sjrd) [inline src/main/scala/ScalaModulePlugin.scala]: Shouldn't `mimaPreviousVersion` also be moved to `scalaModuleSettings`?

Comment (lrytz) [inline src/main/scala/ScalaModulePlugin.scala]: right, thanks! so hot in this office...

**Comment by lrytz** [comment]
Thanks - still got to fix some stuff to make MiMa actually work on the js artifacts. Testing that locally.

File: src/main/scala/ScalaModulePlugin.scala (hunk 1 above)
Comment (dwijnand) [inline src/main/scala/ScalaModulePlugin.scala]: 🎾

**Comment by lrytz** [comment]
@sjrd @dwijnand I have MiMa working now for both jvm and js projects. The changes here are small, but they add the Scala.js sbt plugin here to make `%%%` available.

Here's a log of using this plugin with scala-parser-combinators, it works correctly. MiMa would have caught https://github.com/scala/scala-parser-combinators/issues/119 before the release of 1.0.6.

```
> show mimaPreviousClassfiles
[info] scala-parser-combinatorsJS/*:mimaPreviousClassfiles
[info] 	Map(org.scala-lang.modules:scala-parser-combinators_sjs0.6_2.12:1.0.5 -> /Users/luc/.ivy2/cache/org.scala-lang.modules/scala-parser-combinators_sjs0.6_2.12/jars/scala-parser-combinators_sjs0.6_2.12-1.0.5.jar)

[info] scala-parser-combinatorsJVM/*:mimaPreviousClassfiles
[info] 	Map(org.scala-lang.modules:scala-parser-combinators_2.12:1.0.5 -> /Users/luc/.ivy2/cache/org.scala-lang.modules/scala-parser-combinators_2.12/bundles/scala-parser-combinators_2.12-1.0.5.jar)

> scala-parser-combinatorsJS/mimaReportBinaryIssues
[info] scala-parser-combinators: found 0 potential binary incompatibilities while checking against org.scala-lang.modules:scala-parser-combinators_sjs0.6_2.12:1

> scala-parser-combinatorsJVM/mimaReportBinaryIssues
[info] scala-parser-combinators: found 0 potential binary incompatibilities while checking against org.scala-lang.modules:scala-parser-combinators_2.12:1.0.5

> set every mimaCheckDirection := "forward"

> scala-parser-combinatorsJS/mimaReportBinaryIssues
[info] scala-parser-combinators: found 2 potential binary incompatibilities while checking against org.scala-lang.modules:scala-parser-combinators_sjs0.6_2.12:1.0.5
[error]  * class scala.util.parsing.json.package does not have a correspondent in other version
[error]    filter with: ProblemFilters.exclude[MissingClassProblem]("scala.util.parsing.json.package")
[error]  * object scala.util.parsing.json.package does not have a correspondent in other version
[error]    filter with: ProblemFilters.exclude[MissingClassProblem]("scala.util.parsing.json.package$")

> scala-parser-combinatorsJVM/mimaReportBinaryIssues
[info] scala-parser-combinators: found 2 potential binary incompatibilities while checking against org.scala-lang.modules:scala-parser-combinators_2.12:1.0.5
[error]  * class scala.util.parsing.json.package does not have a correspondent in other version
[error]    filter with: ProblemFilters.exclude[MissingClassProblem]("scala.util.parsing.json.package")
[error]  * object scala.util.parsing.json.package does not have a correspondent in other version
[error]    filter with: ProblemFilters.exclude[MissingClassProblem]("scala.util.parsing.json.package$")


// correct: package object `json` was added between 1.0.5 and 1.0.6
// https://github.com/scala/scala-parser-combinators/commit/7815fd9abbe7b26c3bdd7772249c3e070175976e


> set every ScalaModulePlugin.mimaPreviousVersion := Some("1.0.6")

// ok, the current master is forward-compatible with 1.0.6
> scala-parser-combinatorsJVM/mimaReportBinaryIssues
[info] scala-parser-combinators: found 0 potential binary incompatibilities while checking against org.scala-lang.modules:scala-parser-combinators_2.12:1.0.6


// ok, all the stuff that's missing the sjs-1.0.6 jar shows as errors
> scala-parser-combinatorsJS/mimaReportBinaryIssues
[info] scala-parser-combinators: found 77 potential binary incompatibilities while checking against org.scala-lang.modules:scala-parser-combinators_sjs0.6_2.12:1.0.6
[error]  * object scala.util.parsing.input.StreamReader does not have a correspondent in other version
[error]    filter with: ProblemFilters.exclude[MissingClassProblem]("scala.util.parsing.input.StreamReader$")
[error]  * object scala.util.parsing.input.CharArrayReader does not have a correspondent in other version
[error]    filter with: ProblemFilters.exclude[MissingClassProblem]("scala.util.parsing.input.CharArrayReader$")
...
[error] (scala-parser-combinatorsJS/*:mimaReportBinaryIssues) scala-parser-combinators: Binary compatibility check failed!
```

File: src/main/scala/ScalaModulePlugin.scala [hunk 2]
```diff
@@ -176,8 +174,12 @@ object ScalaModulePlugin extends AutoPlugin {
   private val runMimaIfEnabled = taskKey[Unit]("Run MiMa if mimaPreviousVersion and the module can be resolved against the current scalaBinaryVersion.")
 
   private lazy val mimaSettings: Seq[Setting[_]] = MimaPlugin.mimaDefaultSettings ++ Seq(
-    // manual cross-versioning because https://github.com/typesafehub/migration-manager/issues/62
-    mimaPreviousArtifacts := Set(organization.value % s"${name.value}_${scalaBinaryVersion.value}" % mimaPreviousVersion.value.getOrElse("dummy")),
+    mimaPreviousVersion := None,
+
+    {
+      import org.scalajs.sbtplugin.ScalaJSPlugin.autoImport._
+      mimaPreviousArtifacts := Set(organization.value %%% s"${moduleName.value}" % mimaPreviousVersion.value.getOrElse("dummy"))
```
Comment (sjrd) [inline src/main/scala/ScalaModulePlugin.scala]: I think you should be able to use the following, which won't require the dependency on `sbt-scalajs`:
```scala
mimaPreviousArtifacts := Set(organization.value % moduleName.value % mimaPreviousVersion.value.getOrElse("dummy") cross crossVersion.value)
```

Comment (lrytz) [inline src/main/scala/ScalaModulePlugin.scala]: Thanks, I'll try that!

Comment (lrytz) [inline src/main/scala/ScalaModulePlugin.scala]: Works! I pushed that change.

**Comment by lrytz** [comment]
Will release 1.0.11 now.

**Pull request merged**